app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,46 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedApiService.Companion.THUMBNAILS_URL
+import com.lightricks.feedexercise.network.TemplateMetadata
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+class FeedRepository(
+    private val feedApiService: FeedApiService = FeedApiService.create()) {
+    private val cachedFeedItems = MutableStateFlow<List<FeedItemEntity>>(emptyList())
+
+    fun feedItemsFlow(): Flow<List<FeedItem>> = cachedFeedItems.map { entities ->
+        entities.map { entity ->
+            FeedItem(
+                entity.id,
+                entity.thumbnailUrl,
+                entity.isPremium)
+        }
+    }
+
+    suspend fun refreshFeedItems() {
+        val templatesMetadata: TemplatesMetadata = feedApiService.getTemplatesMetadata()
+        val refreshedFeedItemEntities = templatesMetadata.templates.map { templateMetadata ->
+            feedItemEntityFromTemplateMetadata(templateMetadata)
+        }
+        cachedFeedItems.value = refreshedFeedItemEntities
+    }
+
+    private fun feedItemEntityFromTemplateMetadata(templateMetadata: TemplateMetadata): FeedItemEntity =
+        FeedItemEntity(
+            id = templateMetadata.id,
+            thumbnailUrl = THUMBNAILS_URL + templateMetadata.templateThumbnailURI,
+            isPremium = templateMetadata.isPremium)
```

Comment:
Such method in Kotlin we tend to write them like the following:
```kotlin
fun TemplateMetadata.toFeedItemEntity(): FeedItemEntity
```
This is sure not mandatory, but good to know the syntax. Extension methods are very useful in Kotlin.

Reply:
Nice syntax indeed!

Reply:
FeedItemEntity does not exist in this commit, it appears here because we made a Mish-Mash. So it's impossible to fix this here without creating FeedItemEntity, which will cause further conflicts when rebasing later.
I'll fix this on the 2nd pull request.

Reply:
We should cache then FeedItem nor FeedItemEntity. We should be able to run the code.